publish.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+#
+# Usage:
+#
+#   publish.sh [RESULT_DIR]
+#
+# RESULT_DIR is optional and can be any path (defult: ./result)
+#
+# This script is used to publish all results found in the directory given by
+# the first script argument. Alternatively, if the argument is not given a
+# ./result subdir will be assumed. The script looks for "mike" and "packages"
+# directories an uploads all relevant files sequentially to the MIKELANGELO
+# package hub.
+#
+# Before using this script, you have to configure s3cmd (http://s3tools.org/s3cmd).
+# After installing, you can invoke the configuration command
+#
+# $ s3cmd --configure
+#
+# You will be asked few AWS related questions (region, key, secret) that you need to
+# provide for the user with write access.
+
```

Comment:
Would it make sense to set following bash scripting flags here:
```
set -o errexit
set -o nounset
set -o pipefail
set -o errtrace
```
Having them here the script exits on first error. I'm not sure if all of them are needed (copied them from [Virtlet](https://github.com/Mirantis/virtlet/blob/master/deploy/demo.sh#L3-L6)), but let me demonstrate the difference.

#### Without the flags
```bash
$ ./publish.sh 
Uploading packages from result/
Uploading image osv-loader
Uploading image file result//mike/osv-loader/index.yaml to s3://mikelangelo-capstan/mike/osv-loader/index.yaml
./publish.sh: line 32: s3cmd: command not found
Uploading image file result//mike/osv-loader/osv-loader.qemu.gz to s3://mikelangelo-capstan/mike/osv-loader/osv-loader.qemu.gz
./publish.sh: line 32: s3cmd: command not found
Uploading package file: result//packages/apache.spark.mpm to s3://mikelangelo-capstan/packages/apache.spark.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/apache.spark.yaml to s3://mikelangelo-capstan/packages/apache.spark.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/erlang.mpm to s3://mikelangelo-capstan/packages/erlang.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/erlang.yaml to s3://mikelangelo-capstan/packages/erlang.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/mysql.mpm to s3://mikelangelo-capstan/packages/mysql.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/mysql.yaml to s3://mikelangelo-capstan/packages/mysql.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/node-4.4.5.mpm to s3://mikelangelo-capstan/packages/node-4.4.5.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/node-4.4.5.yaml to s3://mikelangelo-capstan/packages/node-4.4.5.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/node-4.8.2.mpm to s3://mikelangelo-capstan/packages/node-4.8.2.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/node-4.8.2.yaml to s3://mikelangelo-capstan/packages/node-4.8.2.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/node-6.10.2.mpm to s3://mikelangelo-capstan/packages/node-6.10.2.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/node-6.10.2.yaml to s3://mikelangelo-capstan/packages/node-6.10.2.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/ompi.mpm to s3://mikelangelo-capstan/packages/ompi.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/ompi.yaml to s3://mikelangelo-capstan/packages/ompi.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.core.mpm to s3://mikelangelo-capstan/packages/openfoam.core.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.core.yaml to s3://mikelangelo-capstan/packages/openfoam.core.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.pimplefoam.mpm to s3://mikelangelo-capstan/packages/openfoam.pimplefoam.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.pimplefoam.yaml to s3://mikelangelo-capstan/packages/openfoam.pimplefoam.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.pisofoam.mpm to s3://mikelangelo-capstan/packages/openfoam.pisofoam.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.pisofoam.yaml to s3://mikelangelo-capstan/packages/openfoam.pisofoam.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.poroussimplefoam.mpm to s3://mikelangelo-capstan/packages/openfoam.poroussimplefoam.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.poroussimplefoam.yaml to s3://mikelangelo-capstan/packages/openfoam.poroussimplefoam.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.potentialfoam.mpm to s3://mikelangelo-capstan/packages/openfoam.potentialfoam.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.potentialfoam.yaml to s3://mikelangelo-capstan/packages/openfoam.potentialfoam.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.rhoporoussimplefoam.mpm to s3://mikelangelo-capstan/packages/openfoam.rhoporoussimplefoam.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.rhoporoussimplefoam.yaml to s3://mikelangelo-capstan/packages/openfoam.rhoporoussimplefoam.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.rhosimplefoam.mpm to s3://mikelangelo-capstan/packages/openfoam.rhosimplefoam.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.rhosimplefoam.yaml to s3://mikelangelo-capstan/packages/openfoam.rhosimplefoam.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.simplefoam.mpm to s3://mikelangelo-capstan/packages/openfoam.simplefoam.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openfoam.simplefoam.yaml to s3://mikelangelo-capstan/packages/openfoam.simplefoam.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openjdk7.mpm to s3://mikelangelo-capstan/packages/openjdk7.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openjdk7.yaml to s3://mikelangelo-capstan/packages/openjdk7.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openjdk8-zulu-compact1.mpm to s3://mikelangelo-capstan/packages/openjdk8-zulu-compact1.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openjdk8-zulu-compact1.yaml to s3://mikelangelo-capstan/packages/openjdk8-zulu-compact1.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openjdk8-zulu-compact3-with-java-beans.mpm to s3://mikelangelo-capstan/packages/openjdk8-zulu-compact3-with-java-beans.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/openjdk8-zulu-compact3-with-java-beans.yaml to s3://mikelangelo-capstan/packages/openjdk8-zulu-compact3-with-java-beans.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.bootstrap.mpm to s3://mikelangelo-capstan/packages/osv.bootstrap.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.bootstrap.yaml to s3://mikelangelo-capstan/packages/osv.bootstrap.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.cli.mpm to s3://mikelangelo-capstan/packages/osv.cli.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.cli.yaml to s3://mikelangelo-capstan/packages/osv.cli.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.cloud-init.mpm to s3://mikelangelo-capstan/packages/osv.cloud-init.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.cloud-init.yaml to s3://mikelangelo-capstan/packages/osv.cloud-init.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.httpserver.mpm to s3://mikelangelo-capstan/packages/osv.httpserver.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.httpserver.yaml to s3://mikelangelo-capstan/packages/osv.httpserver.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.nfs.mpm to s3://mikelangelo-capstan/packages/osv.nfs.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.nfs.yaml to s3://mikelangelo-capstan/packages/osv.nfs.yaml
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.process-builder.mpm to s3://mikelangelo-capstan/packages/osv.process-builder.mpm
./publish.sh: line 39: s3cmd: command not found
Uploading package file: result//packages/osv.process-builder.yaml to s3://mikelangelo-capstan/packages/osv.process-builder.yaml
./publish.sh: line 39: s3cmd: command not found
DONE. All packages uploaded to S3 repository
```
Man, how I got scared when I've seen `DONE. All packages uploaded to S3 repository`. For a moment I thought I've actually uploaded all the packages somewhere.

#### When using the flags
```bash
$ ./publish.sh 
Uploading packages from result/
Uploading image osv-loader
Uploading image file result//mike/osv-loader/index.yaml to s3://mikelangelo-capstan/mike/osv-loader/index.yaml
./publish.sh: line 37: s3cmd: command not found
```